frontend/src/views/SoilTests/SoilTests.tsx

```diff
@@ -29,30 +31,43 @@ import { InfoBox } from './soilTests.styles';
 import useAppState from '@/hooks/useAppState';
 import { CROPS, FIELD_LIST } from '@/constants/routes';
 import SoilTestsModal from './SoilTestsModal';
+import {
+  sharedCalcCropReq,
+  postprocessModalData,
+} from '@/calculations/FieldAndSoil/Crops/Calculations';
 
 export default function SoilTests() {
   const { state, dispatch } = useAppState();
   const navigate = useNavigate();
   const apiCache = useContext(APICacheContext);
 
-  const [fields, setFields] = useState<NMPFileField[]>(state.nmpFile.years[0].fields || []);
+  const [fields, setFields] = useState<NMPFileField[]>(
+    structuredClone(state.nmpFile.years[0].fields) || [],
+  );
   const [soilTestId, setSoilTestId] = useState<number>(
     fields.find((field) => field.soilTest !== undefined)?.soilTest?.soilTestId || 0,
   );
 
+  const [crops, setCrops] = useState<Crop[]>([]);
+  const [cropTypes, setCropTypes] = useState<CropType[]>([]);
+
   const [soilTestMethods, setSoilTestMethods] = useState<SelectOption<SoilTestMethods>[]>([]);
   const [phosphorousRanges, setPhosphorousRanges] = useState<SoilTestPhosphorousRange[]>([]);
   const [potassiumRanges, setPotassiumRanges] = useState<SoilTestPotassiumRange[]>([]);
   const [currentFieldIndex, setCurrentFieldIndex] = useState<number | null>(null);
 
+  const [fieldsEdited, setFieldsEdited] = useState<number[]>([]);
```

Comment:
Instead of tracking what fields changed, you might just want to recalculate anything that has crops. Needing to track changes seems prone to error. For example, this current code makes updates if a soil test is edited or deleted, but if a soil test is **added** to a field with existing crops, no update is triggered.